swfiles/sw_fitpowder.m

```diff
@@ -0,0 +1,308 @@
+classdef sw_fitpowder < handle & matlab.mixin.SetGet
+    % class to fit powders
+    %
+    % [spinw.copy], [spinw.struct], [Comparing handle and value classes](https://www.mathworks.com/help/matlab/matlab_oop/comparing-handle-and-value-classes.html)
+    %
+
+    properties (SetObservable)
+        % data
+       swobj;
+       y
+       e
+       ebin_cens
+       modQ_cens
+       nQ = 10  % number |Q| bins to calc. in integration limits of 1D cut
+       ndim
+       ncuts = 0;
+       % spinw funtion inputs
+       sw_instrument_args = struct()
+       powspec_args = struct('nRand', 1e3, 'T', 0, 'formfact', true, 'hermit', false,  'imagChk', false, 'fibo', true, 'binType', 'cbin', 'component', 'Sperp')
+       % functions
+       fit_func
+       cost_function = "Rsq";  % "Rsq" or "chisq"
+       background_strategy = "planar" % "planar" or "independent" (1D only - fbg = @(en, p1, p2, ..., pN)
+       fbg
+       fbg_planar = @(en, modQ, slope_en, slope_modQ, intercept) slope_en*en(:) + slope_modQ*modQ(:)' + intercept;
+       fbg_indep = @(en, slope_en, intercept) slope_en*en(:) + intercept;
```

Comment:
I think `fbg_planar` and `fbg_indep` should be a `properties (Constant)` or `properties (SetAccess = private)` rather than lumped with all the other properties...

Reply:
Technically the user could set their own background function e.g. quadratic for multi-phonon?